### PR TITLE
New: Display user-friendly hint names

### LINKED ISSUE #1516: [extension-browser] Display user-friendly hint names
Currently the extension displays the ID of a hint in results. This is much more terse and not always as clear as the name displayed in the documentation.

Requires importing the updated hint metadata from #1480. Also to avoid duplication this should include work to generate the list of hints to import metadata for from the `package.json` of the extension.

### PR DESCRIPTION
**Pull request by antross**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- ~Added/Updated related documentation.~
- ~Added/Updated related tests.~

## Short description of the change(s)

Fix #1516

Also eliminate duplicate references by generating imports for hints
and hint metadata from package.json.

### CODE REVIEW

**Comment by antross** [comment]
Bundle for testing:
* [browser-extension-fix-1516.zip](https://github.com/webhintio/hint/files/2793113/browser-extension-fix-1516.zip)

File: packages/extension-browser/package.json [hunk 1]
```diff
@@ -89,6 +89,7 @@
     "build": "npm run clean && npm-run-all build:*",
     "build-release": "npm run clean && npm run build:assets && tsc --inlineSourceMap false --removeComments true",
     "build:assets": "cpx \"./{src,tests}/**/{!(*.ts),.!(ts)}\" dist",
+    "build:hints": "node ./scripts/import-hints.js",
```
Comment (molant) [inline packages/extension-browser/package.json]: Shouldn't we do this as part of `build-release`?

File: packages/extension-browser/scripts/import-hints.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        return `    ...Object.values<IHintConstructor>(require('${name}'))`;
+    }).join(',\n')
+}
+];
+
+export default hints;
+`;
+
+writeFile(`${__dirname}/../src/shared/hints.import.ts`, hints, (err) => {
+    console.error(err);
```
Comment (molant) [inline packages/extension-browser/scripts/import-hints.js]: I'd a bit more output. Right now the execution of the script outputs which let me 🤔:

```
null
null
```

Maybe do:

```suggestion
    if (err) {
        console.error(err);
    } else {
        console.log(`Created: ${__dirname}/../src/shared/hints.import.ts`);
    }
```
Or just the check for `if (err) { }`.

Comment (molant) [inline packages/extension-browser/scripts/import-hints.js]: If this is going to be part of `build-release` or similar we need to handle the `exitCode` to make sure we not continue the build process if this script failed.

File: packages/extension-browser/scripts/import-hints.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        return `    ...Object.values<HintMetadata>(require('${name}/dist/src/meta'))`;
+    }).join(',\n')
+}
+];
+
+export default metas;
+`;
+
+writeFile(`${__dirname}/../src/shared/metas.import.ts`, metas, (err) => {
+    console.error(err);
```
Comment (molant) [inline packages/extension-browser/scripts/import-hints.js]: Same comments as for the other file 😊

**Comment by molant** [comment]
For those curious this is how it currently looks like:

image

File: packages/extension-browser/scripts/import-hints.js (hunk 2 above)
Comment (antross) [reply packages/extension-browser/scripts/import-hints.js]: What's our preferred way of handling this? Calling `process.exit()`, throwing an exception, or something else?

Comment (antross) [reply packages/extension-browser/scripts/import-hints.js]: Nevermind, `eslint` tells me to throw an error. 😊

**Comment by antross** [comment]
@molant I updated this to `throw` as it both exists the process and gets the error logged to the console in one step. If you strongly disagree let me know. 😉

**Comment by molant** [comment]
>  If you strongly disagree let me know

Not that strongly 😋

**Pull request merged**